[Linux-SGX] Error of process Builder.

<!-- Thanks for reporting a bug.

Please provide as much information as possible.
Paste any logs using three backticks (```).

IMPORTANT: please do not post general questions here;
if you have questions, please use mailing list: support@graphene-project.io
-->

## Description of the problem

Compared with the commit on 1b8848bdae09fa0e92e4644b7fbd4e8c6cd1718c (on Jan 21), for the SGX graphene, the current new master has issues with the ProcessBuilder with Java.

Sample java code
```java=
package com.mycompany.app;

import java.io.File;
import java.io.IOException;

public class Testls{public static void main(String[] args) throws IOException {

    try{
        String[] cmds = {"ls"};
        ProcessBuilder builder = new ProcessBuilder(cmds);
        builder.redirectErrorStream(true);
        File file = new File("ls.log");
        builder.redirectOutput(file);
        Process process = builder.start(); // the error for SGX appears in this line.
        if (process.waitFor() != 0){
            // cannot run
            System.exit(1);
        }
        System.out.println("######## test ls finished ##############\n");



    }catch (IOException ex){
        ex.printStackTrace();
    }
    catch (java.lang.InterruptedException ex){
        ex.printStackTrace();
    }

}
}
```
Outside SGX environment the code can run well without error, but inside the code there will be error, and error for java part is:

```
java.io.IOException: Cannot run program "ls": error=22, Invalid argument
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at com.mycompany.app.Testls.main(Testls.java:14)
Caused by: java.io.IOException: error=22, Invalid argument
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 1 more
```

And the full log is attached here.
[Testls_sgx_logall.log](https://github.com/oscarlab/graphene/files/6513490/Testls_sgx_logall.log)

However on previous commit (on Jan 21) with SGX environments, no such error occurs.

--- a/LibOS/shim/test/regression/large_mmap.c
+++ b/LibOS/shim/test/regression/large_mmap.c
@@ -1,7 +1,9 @@
 #define _GNU_SOURCE
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/mman.h>
 #include <sys/types.h>
+#include <sys/wait.h>
 #include <unistd.h>
 
 #define TEST_LENGTH  0x10000f000
@@ -42,6 +44,29 @@ int main(void) {
     }
     ((char*)a)[0x100000000] = 0xff;
     printf("large_mmap: mmap 2 completed OK\n");
+
+#if 0
+    /* The below fork tests sending of large checkpoints: at this point, the process allocated >4GB
+     * of memory and must send it to the child. Thus, this fork stresses 32-bit/64-bit logic in
+     * Graphene (especially on SGX PAL). However, for SGX enclaves, this takes several minutes to
+     * execute on wimpy machines (with 128MB of EPC), so it is commented out by default. */
+
+    pid_t pid = fork();
+    if (pid < 0) {
+        perror("fork");
+        return 1;
+    } else if (pid > 0) {
+        int status;
+        if (wait(&status) < 0) {
+            perror("wait");
+            return 1;
+        }
+        if (!WIFEXITED(status) || WEXITSTATUS(status) != 0) {
+            fprintf(stderr, "wrong wait() status: %d\n", status);
+            return 1;
+        }
+    }
+#endif
 
     return 0;
 }

--- a/Pal/src/host/Linux-SGX/db_process.c
+++ b/Pal/src/host/Linux-SGX/db_process.c
@@ -291,9 +291,6 @@ static int64_t proc_read(PAL_HANDLE handle, uint64_t offset, uint64_t count, voi
     if (offset)
         return -PAL_ERROR_INVAL;
 
-    if (count != (uint32_t)count)
-        return -PAL_ERROR_INVAL;
-
     ssize_t bytes;
     if (handle->process.ssl_ctx) {
         bytes = _DkStreamSecureRead(handle->process.ssl_ctx, buffer, count,
@@ -308,9 +305,6 @@ static int64_t proc_read(PAL_HANDLE handle, uint64_t offset, uint64_t count, voi
 
 static int64_t proc_write(PAL_HANDLE handle, uint64_t offset, uint64_t count, const void* buffer) {
     if (offset)
-        return -PAL_ERROR_INVAL;
-
-    if (count != (uint32_t)count)
         return -PAL_ERROR_INVAL;
 
     ssize_t bytes;

--- a/Pal/src/host/Linux-SGX/db_sockets.c
+++ b/Pal/src/host/Linux-SGX/db_sockets.c
@@ -410,9 +410,6 @@ static int64_t tcp_read(PAL_HANDLE handle, uint64_t offset, uint64_t len, void* 
     if (handle->sock.fd == PAL_IDX_POISON)
         return 0;
 
-    if (len != (uint32_t)len)
-        return -PAL_ERROR_INVAL;
-
     ssize_t bytes = ocall_recv(handle->sock.fd, buf, len, NULL, NULL, NULL, NULL);
 
     if (bytes < 0)
@@ -431,9 +428,6 @@ static int64_t tcp_write(PAL_HANDLE handle, uint64_t offset, uint64_t len, const
 
     if (handle->sock.fd == PAL_IDX_POISON)
         return -PAL_ERROR_CONNFAILED;
-
-    if (len != (uint32_t)len)
-        return -PAL_ERROR_INVAL;
 
     ssize_t bytes = ocall_send(handle->sock.fd, buf, len, NULL, 0, NULL, 0);
     if (bytes < 0)
@@ -550,9 +544,6 @@ static int64_t udp_receive(PAL_HANDLE handle, uint64_t offset, uint64_t len, voi
     if (handle->sock.fd == PAL_IDX_POISON)
         return -PAL_ERROR_BADHANDLE;
 
-    if (len != (uint32_t)len)
-        return -PAL_ERROR_INVAL;
-
     ssize_t ret = ocall_recv(handle->sock.fd, buf, len, NULL, NULL, NULL, NULL);
     return ret < 0 ? unix_to_pal_error(ret) : ret;
 }
@@ -567,9 +558,6 @@ static int64_t udp_receivebyaddr(PAL_HANDLE handle, uint64_t offset, uint64_t le
 
     if (handle->sock.fd == PAL_IDX_POISON)
         return -PAL_ERROR_BADHANDLE;
-
-    if (len != (uint32_t)len)
-        return -PAL_ERROR_INVAL;
 
     struct sockaddr_storage conn_addr;
     size_t conn_addrlen = sizeof(conn_addr);
@@ -602,9 +590,6 @@ static int64_t udp_send(PAL_HANDLE handle, uint64_t offset, uint64_t len, const 
     if (handle->sock.fd == PAL_IDX_POISON)
         return -PAL_ERROR_BADHANDLE;
 
-    if (len != (uint32_t)len)
-        return -PAL_ERROR_INVAL;
-
     ssize_t bytes = ocall_send(handle->sock.fd, buf, len, NULL, 0, NULL, 0);
     if (bytes < 0)
         return unix_to_pal_error(bytes);
@@ -624,9 +609,6 @@ static int64_t udp_sendbyaddr(PAL_HANDLE handle, uint64_t offset, uint64_t len, 
         return -PAL_ERROR_BADHANDLE;
 
     if (!strstartswith(addr, URI_PREFIX_UDP))
-        return -PAL_ERROR_INVAL;
-
-    if (len != (uint32_t)len)
         return -PAL_ERROR_INVAL;
 
     addr += static_strlen(URI_PREFIX_UDP);

